plugin/src/test/kotlin/com/github/triplet/gradle/play/CompatibilityTest.kt

```diff
@@ -0,0 +1,121 @@
+) {
+    private val pluginBinaryDir = File("build/libs")
+    private val pluginVersionName = System.getProperty("VERSION_NAME")
+
+    private lateinit var testProject: Project
+
+    @Before
+    fun setup() {
+        testProject = ProjectBuilder.builder()
+                .withProjectDir(File("src/test/fixtures/android_app"))
+                .build()
+    }
+
+    @Test
+    fun pluginIsCompatible() {
+        assert(pluginTest())
+    }
+
+    private fun pluginTest(): Boolean {
+        val pluginJar = pluginBinaryDir
+                .listFiles()
+                .first { it.name.endsWith("$pluginVersionName.jar") }
+                .absolutePath
+
+        File(testProject.projectDir, "build.gradle").writeText("""
+        buildscript {
+            repositories {
+                google()
+                jcenter()
+            }
+            dependencies {
+                classpath 'com.android.tools.build:gradle:$agpVersion'
+                classpath files("$pluginJar")
+
+                // manually defining transitive dependencies for our plugin
+                // as we don't have the pom but only the compiled jar
+                classpath('com.google.apis:google-api-services-androidpublisher:v3-rev12-1.23.0') {
+                    exclude group: 'com.google.guava', module: 'guava-jdk5'
+                }
+            }
+        }
+
+        apply plugin: 'com.android.application'
+        apply plugin: 'com.github.triplet.play'
+
+        android {
+            compileSdkVersion 27
+            buildToolsVersion '27.0.3'
+            lintOptions {
+                abortOnError false
+            }
+
+            defaultConfig {
+                applicationId "com.github.triplet.gradle.play.test"
+                minSdkVersion 21
+                targetSdkVersion 27
+                versionCode 1
+                versionName "1.0"
+            }
+        }
+
+        play {
+            serviceAccountCredentials = file('some-file.json')
+        }
+        """)
+
+        val gradleDist = "https://services.gradle.org/distributions/gradle-$gradleVersion-all.zip"
+
+        GradleRunner.create()
+                .withPluginClasspath()
+                .withGradleDistribution(URI(gradleDist))
+                .withProjectDir(testProject.projectDir)
+                .withArguments("tasks")
+                .build()
+        GradleRunner.create()
+                .withPluginClasspath()
+                .withGradleDistribution(URI(gradleDist))
+                .withProjectDir(testProject.projectDir)
+                .withArguments("clean")
+                .build()
+
+        return true
+    }
+
+    @After
+    fun cleanup() {
+        File(testProject.projectDir, "build.gradle").delete()
+    }
+
+    private companion object {
+        @JvmStatic
+        @Parameterized.Parameters(name = "agpVersion: {0}, gradleVersion {1}")
+        fun parameters() = listOf(
+                // agp 3.0.1, requires at least gradle 4.1
```

Comment:
gradle 4.1 gives me a strange warning of `compileSdkVersion is not specified.` that doesn't happen in production. I'll have to investigate a bit more